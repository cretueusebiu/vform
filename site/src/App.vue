<template>
  <div class="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8">
    <div class="max-w-5xl mx-auto">
      <div class="mt-4 flex items-start">
        <div>
          <div class="text-2xl font-normal text-indigo-600">
            <a href="/">vform</a>
          </div>
          <div class="mt-1 dark:text-gray-300">
            Handle Laravel-Vue forms and validation with ease.
          </div>
        </div>

        <ToggleTheme class="ml-auto my-auto" />
      </div>

      <Badges class="mt-2" />

      <div class="grid grid-cols-4 gap-5 mt-10">
        <div class="col-span-1">
          <Navigation />
        </div>

        <div class="col-span-3">
          <Demo class="mb-4" />

          <div class="panel">
            <Docs />
          </div>
        </div>
      </div>

      <div class="text-center py-4 text-gray-400">
        Created by <a href="https://github.com/cretueusebiu" target="_blank">Cretu Eusebiu</a> âœŒ
      </div>
    </div>
  </div>
  </div>
</template>

<script lang="ts">
import Docs from './../../docs.md'
import Demo from './components/Demo.vue'
import Badges from './components/Badges.vue'
import Navigation from './components/Navigation.vue'
import ToggleTheme from './components/ToggleTheme.vue'

export default {
  name: 'App',

  components: {
    Docs,
    Demo,
    Badges,
    Navigation,
    ToggleTheme
  }
}
</script>
