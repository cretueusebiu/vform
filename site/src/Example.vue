<template>
  <div>
    this is an Examples
    <button @click="click">
      click
    </button>
  </div>
</template>

<script>
export default {
  methods: {
    click () {
      alert('clicked')
    }
  }
}
</script>
