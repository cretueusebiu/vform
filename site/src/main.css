body {
  background-color: #eef1f4;
}
