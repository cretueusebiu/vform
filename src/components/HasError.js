export default {
  name: 'HasError',

  props: {
    form: {
      type: Object,
      required: true
    },

    field: {
      type: String,
      required: true
    }
  }
}
