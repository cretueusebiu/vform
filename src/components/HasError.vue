<template>
  <div v-if="form.errors.has(field)" class="help-block invalid-feedback" v-html="form.errors.get(field)"/>
</template>

<script>
export default {
  name: 'has-error',

  props: {
    form: {
      type: Object,
      required: true
    },

    field: {
      type: String,
      required: true
    }
  }
}
</script>
