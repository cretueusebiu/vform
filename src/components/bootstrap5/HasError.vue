<template>
  <div v-if="form.errors.has(field)" class="invalid-feedback d-block" v-html="form.errors.get(field)" />
</template>

<script>
import Base from './../HasError'

export default {
  extends: Base
}
</script>
