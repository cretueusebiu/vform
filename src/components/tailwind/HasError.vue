<template>
  <div v-if="form.errors.has(field)" class="mt-1 text-sm text-red-600" v-html="form.errors.get(field)" />
</template>

<script>
import Base from './../HasError'

export default {
  extends: Base
}
</script>
