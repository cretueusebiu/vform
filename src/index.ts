import Form from './Form'
import Errors from './Errors'

export {
  Form,
  Errors,
  Form as default
}
