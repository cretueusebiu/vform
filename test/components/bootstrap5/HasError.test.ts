import { mount } from '@vue/test-utils'
import { Form } from './../../../src'
import HasError from './../../../src/components/bootstrap5/HasError.vue'

describe('HasError', () => {
  test('render error message', () => {
    const form = new Form({ username: '' })
    form.errors.set({ username: 'Username is required' })

    const wrapper = mount(HasError, {
      propsData: { form, field: 'username' }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('doesn`t render error message', () => {
    const form = new Form({ username: '' })

    const wrapper = mount(HasError, {
      propsData: { form, field: 'username' }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
