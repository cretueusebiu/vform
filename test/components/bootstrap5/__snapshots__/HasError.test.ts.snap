// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HasError doesn\`t render error message 1`] = `""`;

exports[`HasError render error message 1`] = `<div class="invalid-feedback d-block">Username is required</div>`;
